lib/translate.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+var _ = require('lodash')
+  , parseOneAddress = require('email-addresses').parseOneAddress
+  , version = require('../package.json').version;
+
+var translationRules = [
+  //transactional template
+  [/<%subject%>/, '{{ subject }}'],
+  [/<%body%>/, '{{ body }}'],
+  [/<%asm_global_unsubscribe_url%>/g, '<a href="?" data-msys-unsubscribe="1">unsubs</a>'],
+  //
+  ////marketing templates
+  [/\[(Sender_[A-Za-z0-0]+)]/g, "{{ $1 or '' }}"],
+  //[/\[Sender_Name]/g, "{{ Sender_Name or '' }}"],
+  //[/\[Sender_Address]/g, "{{ Sender_Address or '' }}"],
+  //[/\[Sender_City]/g, "{{ Sender_City or '' }}"],
+  //[/\[Sender_State]/g, "{{ Sender_State or '' }}"],
+  //[/\[Sender_Zip]/g, "{{ Sender_Zip or '' }}"],
+  [/\[Unsubscribe]/g, '<a href="?" data-msys-unsubscribe="1">unsubs</a>'],
+  [/\[Weblink]/g, "{{ Weblink or '' }}"],
+
+  /* ORDERS of following rules are IMPORTANT */
+  [/\[%email%]/g, "{{ address.email }}"],
+  [/\[%([a-zA-Z0-9_]+)%]/g, function (match, fieldName) { //custom field without default value
+    return `{{ ${fieldName.trim()} }}`;
+  }],
+  [/\[%([a-zA-Z0-9_]+)\s+\|(.*[^\s]?)%]/g, function (match, fieldName, defaultValue) { //custom field with default value
+    defaultValue = defaultValue.replace(/(^\"|\')|(\"|\'$)/g, "").trim(); //trim beginning/trailing quotes
+    return `{{ ${fieldName.trim()} or '${defaultValue}' }}`;
+  }]
+];
+
+/**
+ * Find custom delimited substitution tags. They can be like :tag:, %tag%, #tag# or :tag. It can't contain spaces
+ * We'll respect one type at a time.
+ * In case :tag, we'll find until word boundary.
+ * Valid chars: alphanum, _ (underscore), - (dash)
+ * @param content
+ * @param substitutions
+ * @returns {*}
+ */
+function translateCustomDelimited(content, opts) {
+  if (_.isEmpty(opts)) {
+    return content;
+  }
+
+  let delimiterRegex;
+  //Todo: Fix replacing ::tag (with :{{ tag }}) or ::tag:: (with :{{ tag }}: )
+  if (opts.endingDelimiter) {
+    delimiterRegex = new RegExp(`(\\${opts.beginDelimiter}{1}(\\w+)\\${opts.endingDelimiter}{1})`, 'gi');
```

Comment:
should they be sensitive? but it should not probably have any affect here due to `w+`.